### ISSUE TITLE
natbd: unknown error: 201

### ISSUE THREAD

**Issue by feinerer** [alfredh/baresip#291]
I have following problem with module natbd:

```
natbd: Enable NAT Behavior Discovery using STUN server stun.ekiga.net
natbd: resolved STUN-server for UDP -- 217.10.68.152:3478
natbd: resolved STUN-server for TCP -- 217.10.68.152:3478
natbd_filtering: Test I: stun_response_handler: Unknown error: 201
natbd: NAT filtering failed (Unknown error: 201)
natbd_mapping: stun_response_handler: (Unknown error: 201)
natbd: NAT mapping failed (Unknown error: 201)
natbd_lifetime: stun_response_handler X: Unknown error: 201
natbd: nat_lifetime_handler: (Unknown error: 201)
NAT Binding lifetime for UDP: min=1 cur=2 max=3
natbd: Generic ALG detection failed: Unknown error: 201
```

This is with baresip v0.5.4 on OpenBSD. Firewall (pf) is deactivated for the test. The actual stun server makes no difference (tested with stun.ekiga.net and stun.counterpath.com).

**Comment by alfredh** [comment]
can you try `stun.creytiv.com` which supports "new" STUN ?

**Comment by feinerer** [comment]
That changes the behaviour:

```
natbd: Enable NAT Behavior Discovery using STUN server stun.creytiv.com
natbd: resolved STUN-server for UDP -- 208.68.208.201:3478
natbd: resolved STUN-server for TCP -- 208.68.208.201:3478
Generic ALG for UDP changed from (Unknown) to (Not Detected)
NAT Mapping UDP changed from (Unknown) to (Endpoint Independent)
Generic ALG for TCP changed from (Unknown) to (Not Detected)
NAT Mapping TCP changed from (Unknown) to (Endpoint Independent)
NAT Binding lifetime for UDP: min=3 cur=6 max=0
```

So the problem is only with some STUN servers. For these however the natbd behaviour seems problematic as natbd spams following status messages

```
natbd_lifetime: stun_response_handler X: Unknown error: 201
natbd: nat_lifetime_handler: (Unknown error: 201)
NAT Binding lifetime for UDP: min=1 cur=1 max=1
```

continuously (several per second).

**Comment by alfredh** [comment]
Hi,

for this to work the STUN Server used must be compliant with RFC 5780.

restund supports this version. On the STUN server you also need to have
2 different public IPv4 addresses for the detection process to work.

you should probably check with the stun.ekiga.net provider which versions
they support.

/Alfred
